frontend/catalyst/jax_primitives.py

```diff
@@ -662,6 +662,15 @@ def _grad_abstract(*args, jaxpr, fn, grad_params):
     return tuple(transformed_signature.get_results())
 
 
+def _get_call_jaxpr(jaxpr):
+    """Extracts the `call_jaxpr` from a JAXPR if it exists.""" ""
+    for eqn in jaxpr.eqns:
+        primitive = eqn.primitive
+        if primitive is func_p:
+            return eqn.params.get("call_jaxpr")
```

Comment:
I think `dict.get("key")` will silently return None when the key is not present vs `dict["key"]` will actively bark a KeyError. I know it isn't really likely to happen but if jax decides to remove the `"call_jaxpr"` key of the func primitive then we should actively fail here.